src/dd/dd.rs

```diff
@@ -0,0 +1,640 @@
+            OutFileType::Stdout(ref mut s) => s.write(buf),
+        }
+    }
+}
+
+pub fn uumain(args: Vec<String>) -> i32 {
+    let mut cf = parse_args();
+    //    println!("{}", cf);
+    setup(&mut cf);
+    0
+    //    let signal = chan_signal::notify(&[Signal::INT]);
+    //    let (s_done, r_done) = chan::sync(0);
+    //
+    //    let cf = parse_args();
+    //    let cf2 = Arc::clone(&cf);
+    //
+    //    let handle = thread::spawn(move || run(cf2, s_done));
+    //
+    //    chan_select! { // FIXME: resume copying after printing progress
+    //        signal.recv() -> signal => display_progress(Arc::clone(&cf)),
+    //        r_done.recv() => ()
+    //    }
+    //    handle.join();
+}
+
+pub fn get_u(val: &str) -> Option<usize> {
+    if val.is_empty() || (val.len() == 1 && !val.chars().nth(0).unwrap().is_digit(10))
+        || val.chars().nth(0).unwrap() == '-'
+        {
+            return None;
+        }
+
+    let mut pref = val;
+    let mut size: usize = match val.chars().nth(val.len() - 1) {
+        Some('k') => {
+            pref = &val[..val.len() - 1];
+            1024
+        }
+        Some('b') => {
+            pref = &val[..val.len() - 1];
+            512
+        }
+        Some(m) if !m.is_digit(10) && m != 'k' && m != 'b' => {
+            return None;
+        }
+        _ => 1,
+    };
+
+    if pref.is_empty() || pref.chars().nth(pref.len() - 1).unwrap() == 'x'
+        || pref.chars().nth(0).unwrap() == 'x'
+        {
+            return None;
+        }
+    for m in pref.split('x').collect::<Vec<&str>>() {
+        if let Ok(n) = m.parse::<usize>() {
+            size *= n;
+        } else {
+            return None;
+        }
+    }
+
+    Some(size)
+}
+
+pub fn get_i(val: &str) -> Option<isize> {
+    if val.is_empty() || (val.len() == 1 && !val.chars().nth(0).unwrap().is_digit(10)) {
+        return None;
+    }
+
+    let mut pref = val;
+    let mut size: isize = match val.chars().nth(val.len() - 1) {
+        Some('k') => {
+            pref = &val[..val.len() - 1];
+            1024
+        }
+        Some('b') => {
+            pref = &val[..val.len() - 1];
+            512
+        }
+        Some(m) if !m.is_digit(10) && m != 'k' && m != 'b' => {
+            return None;
+        }
+        _ => 1,
+    };
+
+    if pref.is_empty() || pref.chars().nth(pref.len() - 1).unwrap() == 'x'
+        || pref.chars().nth(0).unwrap() == 'x'
+        {
+            return None;
+        }
+    for m in pref.split('x').collect::<Vec<&str>>() {
+        if let Ok(n) = m.parse::<isize>() {
+            size *= n;
+        } else {
+            return None;
+        }
+    }
+
+    Some(size)
+}
+
+pub fn err<S: Into<String>>(s: S) {
+    eprintln!("{}", s.into());
+    ::std::process::exit(1);
+}
+
+pub fn display_stat(cf: &Conf) {
+    eprintln!(
+        "{}+{} records in\n{}+{} records out",
+        cf.inp_full, cf.inp_part, cf.out_full, cf.out_part
+    );
+
+    match cf.trunc {
```

Comment:
In this case, I think it makes more sense to just do:
```rust
if cf.trunc != 0 {
    eprintln!(
        "{} truncated record{}",
        cf.trunc,
        if cf.trunc == 1 { "" } else { "s" }
    );
}
```

Reply:
Did.